packages/atomic/src/components/commerce/search-box-suggestions/atomic-commerce-search-box-recent-queries/atomic-commerce-search-box-recent-queries.tsx

```diff
@@ -0,0 +1,195 @@
+import {
+  SearchBox,
+  RecentQueriesList,
+  buildRecentQueriesList,
+} from '@coveo/headless/commerce';
+import {Component, Element, Prop, State, h} from '@stencil/core';
+import Clock from '../../../../images/clock.svg';
+import {SafeStorage, StorageItems} from '../../../../utils/local-storage-utils';
+import {once} from '../../../../utils/utils';
+import {
+  getPartialRecentQueryClearElement,
+  getPartialRecentQueryElement,
+  RecentQueriesContainer,
+  RecentQueryClear,
+  RecentQueryIcon,
+  RecentQueryText,
+} from '../../../common/suggestions/recent-queries';
+import {
+  dispatchSearchBoxSuggestionsEvent,
+  SearchBoxSuggestionElement,
+  SearchBoxSuggestions,
+  SearchBoxSuggestionsBindings,
+} from '../../../common/suggestions/suggestions-common';
+import {CommerceBindings as Bindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+
+/**
+ * The `atomic-commerce-search-box-recent-queries` component can be added as a child of an `atomic-commerce-search-box` component, allowing for the configuration of recent query suggestions.
+ *
+ * @part recent-query-title - The 'Recent queries' title.
+ * @part recent-query-clear - The 'Clear' button for clearing recent queries    .
+ *
+ * @internal
+ */
+@Component({
+  tag: 'atomic-commerce-search-box-recent-queries',
+  shadow: true,
+})
+export class AtomicSearchBoxRecentQueries {
```

Comment:
Do we want to include "Commerce" in the class name? I.e., AtomicCommerceSearchBoxRecentQueries

I see that this is what @fpbrault did for the result list.